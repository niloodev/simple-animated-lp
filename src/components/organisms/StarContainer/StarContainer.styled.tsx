import { device } from "@/utils/breakpoints";
import { motion } from "motion/react";
import styled, { css } from "styled-components";

export const StarStaticBackground = styled.div`
  z-index: -50;
  position: absolute;
  width: 100%;
  height: 95%;
  left: 0;
  right: 0;
  margin: auto;
  top: 0;
  bottom: 0;
  overflow: hidden;
`;

export const Star = styled(motion.span)<{ $bottomStar?: boolean }>`
  position: absolute;

  ${(props) =>
    props.$bottomStar
      ? css`
          bottom: 0;
        `
      : css`
          top: 0;
        `}

  left: -30vw;
  display: block;
  height: 0.2rem;
  width: 4.8rem;
  border-radius: 2.4rem;
  background: linear-gradient(
      270deg,
      #b49aff 0.61%,
      rgba(180, 154, 255, 0) 90.89%
    ),
    linear-gradient(0deg, rgba(255, 255, 255, 0.05), rgba(255, 255, 255, 0.05));
  box-shadow: 0 0 1.3rem 0px #b49affbf;

  ${device.desktop} {
    width: 12.3rem;
  }

  &::before {
    content: "";
    position: absolute;
    right: 0;
    top: -0.1rem;
    bottom: 0;
    display: block;
    width: 50%;
    border-radius: 100%;
    height: 0.5rem;
    filter: blur(1.2rem);
    background-color: #b49affbf;
  }
`;
