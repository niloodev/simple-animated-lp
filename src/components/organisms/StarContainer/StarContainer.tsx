"use client";
import { Star, StarStaticBackground } from "./StarContainer.styled";
import { ComponentProps } from "react";

export const StarContainer = (
  props: ComponentProps<typeof StarStaticBackground>
) => {
  return (
    <StarStaticBackground {...props}>
      <Star
        animate={{
          x: ["-30vw", "160vw"],
          scale: [1, 3],
        }}
        transition={{ repeat: Infinity, duration: 5 }}
      />
      <Star
        $bottomStar
        animate={{
          x: ["-30vw", "160vw"],
          scale: [1, 3],
        }}
        transition={{ repeat: Infinity, duration: 5, delay: 2 }}
      />
    </StarStaticBackground>
  );
};
