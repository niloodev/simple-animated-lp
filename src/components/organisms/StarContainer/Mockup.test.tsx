import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { StarContainer } from "./StarContainer";

describe("Organisms: StarContainer", () => {
  it("should render correctly", () => {
    const { container } = render(<StarContainer />);

    expect(container).toMatchSnapshot();
  });
});
