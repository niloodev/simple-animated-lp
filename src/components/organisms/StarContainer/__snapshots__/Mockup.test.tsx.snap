// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms: StarContainer should render correctly 1`] = `
<div>
  <div
    class="sc-341db452-0 bgksSM"
  >
    <span
      class="sc-341db452-1 eLGigt"
    />
    <span
      class="sc-341db452-1 dHaYYN"
    />
  </div>
</div>
`;
